tests/models/test_infer_pip_requirements.py

```diff
@@ -0,0 +1,169 @@
+
+    pipeline = Pipeline([("model", model)])
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(pipeline, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = mlflow.infer_pip_requirements(model_uri, mlflow.sklearn.FLAVOR_NAME)
+    assert _get_pinned_requirement("xgboost") in pip_reqs
+    assert _get_pinned_requirement("scikit-learn") in pip_reqs
+
+
+@required_modules("lightgbm")
+def test_infer_pip_requirements_lightgbm():
+    import lightgbm as lgb
+
+    assert "scikit-learn" in _get_install_requires("lightgbm")
+
+    model = lgb.LGBMClassifier()
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(model, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = mlflow.infer_pip_requirements(model_uri, mlflow.sklearn.FLAVOR_NAME)
+    assert _get_pinned_requirement("lightgbm") in pip_reqs
+    assert _get_pinned_requirement("scikit-learn") not in pip_reqs
+
+    pipeline = Pipeline([("model", model)])
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(pipeline, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = mlflow.infer_pip_requirements(model_uri, mlflow.sklearn.FLAVOR_NAME)
+    assert _get_pinned_requirement("lightgbm") in pip_reqs
+    assert _get_pinned_requirement("scikit-learn") not in pip_reqs
+
+
+@required_modules("catboost")
+def test_infer_pip_requirements_catboost():
+    import catboost
+
+    assert "scikit-learn" not in _get_install_requires("catboost")
+
+    model = catboost.CatBoostClassifier()
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(model, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = mlflow.infer_pip_requirements(model_uri, mlflow.sklearn.FLAVOR_NAME)
+    assert _get_pinned_requirement("catboost") in pip_reqs
+    assert _get_pinned_requirement("scikit-learn") not in pip_reqs
+
+    pipeline = Pipeline([("model", model)])
+    with mlflow.start_run():
+        mlflow.sklearn.log_model(pipeline, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = mlflow.infer_pip_requirements(model_uri, mlflow.sklearn.FLAVOR_NAME)
+    assert _get_pinned_requirement("catboost") in pip_reqs
+    assert _get_pinned_requirement("scikit-learn") in pip_reqs
+
+
+@required_modules("torch", "torchvision", "transformers")
+def test_infer_pip_requirements_transformers_pytorch():
+    from transformers import BertModel, BertConfig
+
+    model = BertModel(BertConfig())
+    with mlflow.start_run():
+        mlflow.pytorch.log_model(model, artifact_path="model")
+        model_uri = mlflow.get_artifact_uri("model")
+
+    pip_reqs = mlflow.infer_pip_requirements(model_uri, mlflow.pytorch.FLAVOR_NAME)
+    assert _get_pinned_requirement("torch") in pip_reqs
+    assert _get_pinned_requirement("transformers") in pip_reqs
```

Comment:
@dbczumar 

The inferred pip requirements of a pytorch-transformers model look like this:

https://github.com/mlflow/mlflow/pull/4518/checks?check_run_id=3153161565#step:6:882

```
tests/models/test_infer_pip_requirements.py::test_infer_pip_requirements_transformers_pytorch 
==============================
Keras-Preprocessing==1.1.2
Pillow==8.3.1
PyYAML==5.4.1
adal==1.2.7
bcrypt==3.2.0
boto3==1.18.6
google-cloud-storage==1.41.1
ipython==7.16.1
kubernetes==17.17.0
pandas==1.1.5
pyOpenSSL==20.0.1
scipy==1.5.4
tensorflow==2.4.2
torch==1.9.0
transformers==4.9.0
==============================
```